### PR TITLE
[Console] Add getters for Application::$autoExit and $catchExceptions

### PR DESCRIPTION
**Pull request by VasekPurchart**
There is currently no way to retrieve this information from Application (afaik) and since this can be set from outside (and is even part of the `@api`) it should be retrievable.

This might come handy for example when working with an Application instance inside your code - for example when creating a Console exception listener - then you need information about how the application was run.

| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by unkind** [comment]
> for example when creating a Console exception listener - then you need information about how the application was run

What's the use case?

**Comment by VasekPurchart** [comment]
When the console application is run internally, you probably don't want to log anything - or probably at least in a different manner. By internally I mean that your code invokes a command (something like [this](http://symfony.com/doc/current/components/console/introduction.html#calling-an-existing-command)) and in this situation, you want to turn off the autoExit feature and also set catchExceptions to false.

When an exception occurres you don't want to log this exception as error, because it still might be a valid case which is handled by the calling code. Or even if it isn't, then you would log this error twice - from the internal run and also when it reaches the point from which it was really run - like a controller or another console application run.

**Comment by fabpot** [comment]
These two flags should only be used for testing commands, and that's why we don't have any getters. I'm :-1: for adding those methods.

**Comment by VasekPurchart** [comment]
@fabpot what about the cases when you [invoke the application internally](http://symfony.com/doc/current/components/console/introduction.html#calling-an-existing-command)? Tests are just a special example of this. I think that reusing existing commands is super useful, but inability to reason about the application state complicates the tooling around it (the only way right now is to check via reflection or pass the settings somehow separately through a different channel).

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -255,6 +255,18 @@ public function getHelp()
     }
 
     /**
+     * Gets whether to catch exceptions or not during commands execution.
+     *
+     * @return bool Whether to catch exceptions or not during commands execution
+     *
+     * @api
+     */
+    public function getCatchExceptions()
```
Comment (fabpot) [inline src/Symfony/Component/Console/Application.php]: should be `areExceptionsCatched()`

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -255,6 +255,18 @@ public function getHelp()
     }
 
     /**
+     * Gets whether to catch exceptions or not during commands execution.
+     *
+     * @return bool Whether to catch exceptions or not during commands execution
+     *
+     * @api
```
Comment (fabpot) [inline src/Symfony/Component/Console/Application.php]: `@api` tags must be removed

File: src/Symfony/Component/Console/Application.php [hunk 3]
```diff
@@ -267,6 +279,18 @@ public function setCatchExceptions($boolean)
     }
 
     /**
+     * Gets whether to automatically exit after a command execution or not.
+     *
+     * @return bool Whether to automatically exit after a command execution or not
+     *
+     * @api
+     */
+    public function getAutoExit()
```
Comment (fabpot) [inline src/Symfony/Component/Console/Application.php]: `isAutoExitEnabled`?

**Comment by fabpot** [comment]
:+1: for master

**Comment by VasekPurchart** [comment]
@fabpot thanks for resurfacing this :) I made the chanes you proposed, although I am not sure about the renamings, I tried for them to be as close to the respective setters as possible, which certainly no longer holds with `setCatchExceptions` vs `areExceptionsCatched` (even if you guess `are`, the word order is different). But if this is in alignment with other SF APIs, then it's of course, just wasn't sure.

Btw since this PR is so old, it was created against `2.7`, should I try to open it against another version or `master` as you wrote? I could easily rebase this although it seems there is no way to change the branch here on GitHub.

**Comment by VasekPurchart** [comment]
@fabpot thanks for resurfacing this :) I made the chanes you proposed, although I am not sure about the renamings, I tried for them to be as close to the respective setters as possible, which certainly no longer holds with `setCatchExceptions` vs `areExceptionsCatched` (even if you guess `are`, the word order is different). But if this is in alignment with other SF APIs, then it's of course, just wasn't sure.

Btw since this PR is so old, it was created against `2.7`, should I try to open it against another version or `master` as you wrote? I could easily rebase this although it seems there is no way to change the branch here on GitHub.

File: src/Symfony/Component/Console/Application.php [hunk 4]
```diff
@@ -255,6 +255,16 @@ public function getHelp()
     }
 
     /**
+     * Gets whether to catch exceptions or not during commands execution.
+     *
+     * @return bool Whether to catch exceptions or not during commands execution
+     */
+    public function areExceptionsCatched()
```
Comment (jakzal) [inline src/Symfony/Component/Console/Application.php]: shouldn't this be `areExceptionsCaught()`?

Comment (fabpot) [inline src/Symfony/Component/Console/Application.php]: indeed

**Comment by fabpot** [comment]
@VasekPurchart Can you fix the typo?

**Comment by VasekPurchart** [comment]
@fabpot sure, it's fixed. Should I rebase this somewhere else, as I asked above, or is it fine here?

**Comment by fabpot** [comment]
@VasekPurchart Can you fix the fabbot issues? No need to rebase, the PR is fine as is.

**Comment by VasekPurchart** [comment]
@fabpot even if they are completely unrelated to this issue/PR? I thought this would be confusing for the diff. I can do it in a separate commit probably.

**Comment by VasekPurchart** [comment]
Fixed.

**Comment by fabpot** [comment]
Thank you @VasekPurchart.

**Comment by VasekPurchart** [comment]
Thanks for merging!

**Pull request closed without merging**